### PR TITLE
fix(slack): span-aware splitter for converted mrkdwn (#25569 follow-up)

### PR DESCRIPTION
**Pull request by siddseethepalli**
## Summary
- PR #25569 reordered `textToSlackBlocks`'s text branch to convert markdown → mrkdwn before calling `splitLongTextSegment`, fixing orphan `](` / `**` tokens. Bot reviewers (Codex, Devin) flagged that this still leaves a residual edge case: link text inside a `<url|...>` span can legitimately contain `. `/`! `/`? ` (e.g. `<url|First sentence. Second sentence>`), and a `<...>` span can straddle the maxChars cutoff with no other in-window delimiters — both produce chunks ending with an unclosed `<` and orphan `>`.
- Makes `splitLongTextSegment` span-aware: `computeMrkdwnSpans` finds `<...>` (including unclosed-at-window-edge) and `*...*` regions; `findBoundary` rejects candidate split points strictly inside a span; the hard-slice fallback backs up to the start of any straddling span.
- Updates the comment on the conversion-then-split block to drop the (factually wrong) 'safe' claim.
- Adds two regression tests covering both edge cases the bot reviewers reproduced (link span with internal `. ` as the only in-window sentence delimiter; link span straddling maxChars with no in-window delimiters).

## Test plan
- [x] `bun test src/runtime/__tests__/slack-block-formatting.test.ts` (20 pass)
- [x] `bunx tsc --noEmit`

---

<a href="https://app.devin.ai/review/vellum-ai/vellum-assistant/pull/25629" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open with Devin
  </picture>
</a>

### CODE REVIEW

**Comment by siddseethepalli** [comment]
Addressed in https://github.com/vellum-ai/vellum-assistant/pull/25653

File: assistant/src/runtime/slack-block-formatting.ts [hunk 1]
```diff
@@ -380,6 +389,64 @@ export function splitLongTextSegment(
...
+  while (i < window.length) {
+    const open = window.indexOf("<", i);
+    if (open < 0) break;
+    const close = window.indexOf(">", open + 1);
+    // For unclosed `<...` (span continues past the window), use a sentinel
+    // larger than any in-window position so split-point checks treat the
+    // window's right edge as inside the span and back the hard-slice up.
+    const end = close < 0 ? window.length + 1 : close + 1;
+    intervals.push([open, end]);
+    i = close < 0 ? window.length : end;
```
[the remark this answers is not in the document]
Comment (siddseethepalli) [reply assistant/src/runtime/slack-block-formatting.ts:412]: Addressed in https://github.com/vellum-ai/vellum-assistant/pull/25653
